### ISSUE TITLE
Error handling in EventStream subscriptions

### ISSUE THREAD

**Issue by rogeralsing** [AsynkronIT/protoactor-dotnet#102]
We need proper error handling in the event stream handlers.
If the code below fails, that might kill the application

```csharp
        public void Publish(T msg)
        {
            foreach (var sub in _subscriptions)
            {
                sub.Value(msg);
            }
        }
```

**Comment by tomliversidge** [comment]
How should errors here be handled? Also, is it right that it is synchronously going through each subscription? Any reason it's not asynchronous, fire and forget?

**Comment by rogeralsing** [comment]
Fixed in #150
